File: mybatis-plus-join-test/src/main/java/com/github/yulichang/test/dto/AddressDTO.java
package com.github.yulichang.test.dto;

import com.github.yulichang.test.entity.AreaDO;
import lombok.Data;

import java.util.List;

@Data
public class AddressDTO {

    private Integer id;

    private Integer userId;

    private Integer areaId;

    private String tel;

    private String address;

    private Boolean del;

    private List<AreaDO> areaList;

    private AreaDO area;
}

File: mybatis-plus-join-test/test-kotlin/src/main/kotlin/com/github/yulichang/test/kt/dto/AddressDTO.kt
package com.github.yulichang.test.kt.dto

import com.github.yulichang.test.kt.entity.AreaDO

class AddressDTO {
    val id: Int? = null
    val userId: Int? = null
    val areaId: Int? = null
    val tel: String? = null
    val address: String? = null
    val del: Boolean? = null
    val areaList: List<AreaDO>? = null
    val area: AreaDO? = null

    override fun toString(): String {
        return "AddressDTO(id=$id, userId=$userId, areaId=$areaId, tel=$tel, address=$address, del=$del, areaList=$areaList, area=$area)"
    }


}
